src/couch/src/couch_util.erl

```diff
@@ -212,29 +212,36 @@ validate_utf8_fast(B, O) ->
             false
     end.
 
-to_hex(<<Hi:4, Lo:4, Rest/binary>>) ->
-    [nibble_to_hex(Hi), nibble_to_hex(Lo) | to_hex(Rest)];
-to_hex(<<>>) ->
-    [];
+to_hex(Binary) when is_binary(Binary) ->
+    binary_to_list(to_hex_bin(Binary));
 to_hex(List) when is_list(List) ->
     to_hex(list_to_binary(List)).
```

Comment:
Do we need this clause to recurse back into to_hex/1 just to have the binary turned back into a list? It'd change the error a bit if List isn't convertible to binary I guess? This is just me thinking out loud, I don't have a direct preference really.

Reply:
Oh, it does look silly this way. I'd just have it call to_hex_bin/1 directly it would look cleaner that way.